ext/datadog_profiling_native_extension/collectors_thread_context.c

```diff
@@ -439,7 +439,16 @@ static VALUE _native_on_gc_finish(DDTRACE_UNUSED VALUE self, VALUE collector_ins
 
 // This method exists only to enable testing Datadog::Profiling::Collectors::ThreadContext behavior using RSpec.
 // It SHOULD NOT be used for other purposes.
-static VALUE _native_sample_after_gc(DDTRACE_UNUSED VALUE self, VALUE collector_instance) {
+static VALUE _native_sample_after_gc(DDTRACE_UNUSED VALUE self, VALUE collector_instance, VALUE reset_monotonic_to_system_state) {
+  ENFORCE_BOOLEAN(reset_monotonic_to_system_state);
+
+  struct thread_context_collector_state *state;
+  TypedData_Get_Struct(collector_instance, struct thread_context_collector_state, &thread_context_collector_typed_data, state);
+
+  if (reset_monotonic_to_system_state == Qtrue) {
+    state->time_converter_state = (monotonic_to_system_epoch_state) MONOTONIC_TO_SYSTEM_EPOCH_INITIALIZER;
+  }
```

Comment:
Can't seem to find any code overriding this state. Why do we need to do this?

Reply:
See commit description for c8f2338b201230849c6cc255c565eed06dc5d214 -- it's my attempt at fixing a flaky spec that showed up.

Reply:
Ack! (not used to seeing useful commit descriptions 😂 )

Reply:
> Ack! (not used to seeing useful commit descriptions 😂 )

🔥🔥🔥